modules/dhcp/providers/server/isc.rb

```diff
@@ -149,11 +149,24 @@ def loadSubnetData subnet
 
     def parse_config_for_subnets
       ret_val = []
-
-      @config.scan(/subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)/) do |match|
-        next unless managed_subnet? "#{match[0]}/#{match[1]}"
-
-        ret_val << Proxy::DHCP::Subnet.new(match[0], match[1])
+      @config.scan(/subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)\s*{([\w|;|\.|,|\-|\s]*)}/) do |match|
+        options = {}
+        if match[2]
+          match[2].split(';').each do |option|
+            case option
+              when /option\s+routers\s+[\d\.]+/
+                # get the first router match and use it for gateway
+                options[:gateway] = option.scan(/option\s+routers\s+([\d\.]+)/).first.first
+              when /option\s+domain\-name\-servers\s+[\d\.]+/
+                # get #1 and #2 domain name servers
+                options[:dns_primary], options[:dns_secondary] = option.scan(/option\s+domain\-name\-servers\s+([\d\.]+)\s*[,]?\s*([\d\.]*)/).first
```

Comment:
similarly, why not keep the whole list? Also, this will leave `:dns_secondary` with an empty string if only one server has been specified.